notion-server/src/ethereum/mod.rs

```diff
@@ -1,25 +1,122 @@
+#![allow(unused)]
+
+use async_std::sync::{channel, Receiver, Sender};
+
+use snafu::{OptionExt, Snafu};
+
 use std::collections::HashMap;
 
+/// Shorthand for result types returned from the Simulation simulation.
+pub type Result<V, E = Error> = std::result::Result<V, E>;
+
+/// Errors arising from the simulation.
+#[derive(Debug, Snafu)]
+pub enum Error {
+    /// Operation was cancelled because the simulation is shutting down.
+    Terminated,
+}
+
+#[derive(Debug)]
+enum Operation {
+    CreateExecutionEnvironment(args::CreateExecutionEnvironment, Sender<EeIndex>),
+    CreateShardChain(args::CreateShardChain, Sender<u32>),
+    CreateShardBlock(args::CreateShardBlock, Sender<u32>),
+    GetShardBlock(args::GetShardBlock, Sender<ShardBlock>),
+}
+
+#[derive(Debug, Clone)]
+pub struct Handle(Sender<Operation>);
+
+impl Handle {
+    pub async fn create_shard_chain(&self, arg: args::CreateShardChain) -> Result<u32> {
```

Comment:
I like that there's a wrapper here -- I think we should shield the internals of the `EthereumSimulation` object from our choice for data sharing / message passing with the server.

Reply:
Yeah, I think all the implementations have a wrapper like this. Calling into async functions without it from Rocket would be a pain.